src/module-server/node-resolve.ts

```diff
@@ -129,7 +129,7 @@ export const resolveFromNodeModules = async (
   const cacheKey = resolveCacheKey(id, importer, root);
   const cached = resolveCache.get(cacheKey);
   if (cached) return cached;
-  const pathChunks = id.split(posix.sep);
+  const pathChunks = id.split(/[/\\]/g);
```

Comment:
Can we add a comment explaining what this regex is looking for and how it works?